server/app/views/NorthStarBaseView.java

```diff
@@ -220,6 +210,40 @@ protected ThymeleafModule.PlayThymeleafContext createThymeleafContext(
     return context;
   }
 
+  private static String getSessionDurationMessage(int sessionDurationMinutes, Messages messages) {
+    int sessionDurationHours = sessionDurationMinutes / 60;
+    // The remaining minutes after accounting for whole hours.
+    int remainingSessionDurationMinutes = sessionDurationMinutes % 60;
+    if (sessionDurationMinutes < 60) {
+      return sessionDurationMinutes == 1
+          ? messages.at(MessageKey.BANNER_MINUTE.getKeyName())
+          : messages.at(
+              MessageKey.BANNER_MINUTES.getKeyName(), String.valueOf(sessionDurationMinutes));
+    }
+    // The session duration is a whole number of hours.
+    if (remainingSessionDurationMinutes == 0) {
+      return sessionDurationHours == 1
+          ? messages.at(MessageKey.BANNER_HOUR.getKeyName())
+          : messages.at(MessageKey.BANNER_HOURS.getKeyName(), String.valueOf(sessionDurationHours));
+    } else {
+      if (sessionDurationHours == 1) {
+        return sessionDurationMinutes == 1
```

Comment:
question (maybe blocking): should this be `remainingSessionDurationMinutes`?

Reply:
Good call updated!